src/Stryker.Core/Stryker.Core/ExcludableString.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Text.RegularExpressions;
+using DotNet.Globbing;
+
+namespace Stryker.Core
+{
+    public readonly struct ExcludableString
+    {
+        private static readonly Regex _mutantSpansRegex = new("(\\{(\\d+)\\.\\.(\\d+)\\})+$");
+
+        public ExcludableString(string s)
+        {
+            if (s is null)
+            {
+                throw new ArgumentNullException(nameof(s));
+            }
+
+            IsExcluded = s.StartsWith('!');
+
+            var pattern = IsExcluded ? s[1..] : s;
+            var mutantSpansRegex = _mutantSpansRegex.Match(pattern);
+            if (mutantSpansRegex.Success)
+            {
+                var filePathPart = pattern[..^mutantSpansRegex.Length];
+                var normalized = FilePathUtils.NormalizePathSeparators(filePathPart);
+                Glob = Glob.Parse(normalized);
+                MutantSpans = mutantSpansRegex.Value;
+            }
+            else
+            {
+                var normalized = FilePathUtils.NormalizePathSeparators(pattern);
+                Glob = Glob.Parse(normalized);
+                MutantSpans = string.Empty;
+            }
+        }
+
+        public bool IsExcluded { get; }
+
+        public Glob Glob { get; }
+
+        public string MutantSpans { get; }
```

Comment:
I feel like MutantSpans might give the wrong impression. Also it's one span right, so shouldn't be plural? Or am I missing something?

Reply:
Regex is noone's cup of tea (maybe apart from Stephen Toub). 

So, there might be several spans, e.g. `{10..19}{20..30}`. [Here in the tests](https://github.com/stryker-mutator/stryker-net/blob/03f991307924b67d14791b7ee42d1f0f1c2c8717/src/Stryker.Core/Stryker.Core.UnitTest/Options/FilePatternTests.cs#L37-L44), you try all sorts of stuff :D And then there is a bunch of logic to reduce, split and intersect the spans.

Now, of course my work in terms of F# context would also simplify if we decide to just remove this. But just from the user perspective, Stryker comments are much more intuitive and powerful, indeed I cannot see a scenario when I would prefer specifying spans over Stryker comments.

Reply:
So it goes down to what you decide and when. If you plan to remove this in the near future, this would kind of eliminate the need of this change. Like, the reasoning I have in the PR description will not apply anymore, FilePattern will reduce to ExclusionPattern, would be used to for the both relevant inputs, and then - eventually - C# files will just assume default full-file `TextSpan`s and F# files will assume default full-file `Range`s.

Reply:
Actually, that would simplify so much stuff than can even volunteer to do that, since I have already spent a lot of time working in this area. But I need your green light :D

Reply:
We need to discuss this first but I'm all for it so I think it's likely we'll do this. The only issue is that since we're removing the feature it's a breaking change and should be in the next major version. I created the issue for it and I'l try to discuss this with @richardwerkman shortly but otherwise we can talk it over during hacktoberfest :)

Reply:
Alright! Well in that case this should be ready for review then and can be taken as a step to the option removal in the future.

Reply:
We'll do this later.